core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/common/JDBCUserIdManager.java

```diff
@@ -0,0 +1,189 @@
+                throw new IdManagerException("Provided user name already exist with an Id.");
+            }
+        } catch (SQLException e) {
+            throw new IdManagerException("Error occurred while performing pre checks.", e);
+        }
+
+        try (Connection connection = dataSource.getConnection()) {
+            PreparedStatement preparedStatement = connection.prepareStatement(SQLQueries.ADD_ID_FOR_NAME);
+            preparedStatement.setString(1, id);
+            preparedStatement.setString(2, name);
+            preparedStatement.execute();
+
+            return new DefaultUser(id, name);
+        } catch (SQLException e) {
+            throw new IdManagerException(e);
+        }
+    }
+
+    @Override
+    public void removeIdForName(Identifiable identifiable) throws IdManagerException {
+
+        if (StringUtils.isEmpty(identifiable.getName()) && StringUtils.isEmpty(identifiable.getId())) {
+            throw new IdManagerException("Identifiable name or id should present.");
+        }
+
+        String name = identifiable.getName();
+        String id = identifiable.getId();
+
+        try (Connection connection = dataSource.getConnection()) {
+            PreparedStatement preparedStatement = connection.prepareStatement(SQLQueries.DELETE_RECORD_FROM_NAME_OR_ID);
+            preparedStatement.setString(1, id);
+            preparedStatement.setString(2, name);
+            preparedStatement.execute();
+        } catch (SQLException e) {
+            throw new IdManagerException(e);
+        }
+    }
+
+    private boolean checkNameExist(String name) throws SQLException {
+
+        try (Connection connection = dataSource.getConnection()) {
+            PreparedStatement preparedStatement = connection.prepareStatement(SQLQueries.CHECK_NAME_EXIST);
+            preparedStatement.setString(1, name);
+            try (ResultSet resultSet = preparedStatement.executeQuery()) {
+                return resultSet.next();
+            }
+        }
+    }
+
+    private boolean checkIdExist(String id) throws SQLException {
+
+        try (Connection connection = dataSource.getConnection()) {
+            PreparedStatement preparedStatement = connection.prepareStatement(SQLQueries.CHECK_ID_EXIST);
+            preparedStatement.setString(1, id);
+            try (ResultSet resultSet = preparedStatement.executeQuery()) {
+                return resultSet.next();
+            }
+        }
+    }
+
+    private static final class SQLQueries {
+
+        private static final String ADD_ID_FOR_NAME = "";
+        private static final String CHECK_NAME_EXIST = "";
+        private static final String CHECK_ID_EXIST = "";
+        private static final String DELETE_RECORD_FROM_NAME_OR_ID = "";
+        private static final String GET_NAME_FROM_ID = "";
+        private static final String GET_ID_FROM_NAME = "";
+    }
+
+    private static final class TableColumns {
+
+        private static final String ID = "";
```

Comment:
Are these strings yet to be decided or are they empty mistakenly?

Reply:
Yet to be decided after the DB schemes finalized.